### PR TITLE
add HList support

### PR DESCRIPTION
**Pull request by agluszyk**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -35,7 +35,10 @@ libraryDependencies ++= Seq(
   "org.scalatestplus.play" %% "scalatestplus-play" % "5.1.0" % Test,
   "com.typesafe.play" %% "play-test" % "2.8.8" % "test",
   "com.typesafe.play" %% "play-json-joda" % "2.9.2",
-  "com.h2database" % "h2" % "1.4.200" % "test"
+  "com.h2database" % "h2" % "1.4.200" % "test",
+  "com.typesafe.slick" %% "slick"     % "3.3.2",
```
Comment (JakubSroka) [inline build.sbt:39]: isn't it already included in unicorn dependency?

File: src/test/scala/org/virtuslab/beholder/form/FormFiltersTestForLargeElements.scala [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+      inText,
+      inText,
+      inText,
+      inText,
+      inText)
+  }.filterGenerator
+
+}
+
+class FormFiltersTestForLargeElements extends BaseTest {
```
Comment (fazzou) [inline src/test/scala/org/virtuslab/beholder/form/FormFiltersTestForLargeElements.scala:52]: maybe test also json filters for large elements?

File: build.sbt (hunk 1 above)
Comment (fazzou) [inline build.sbt:39]: why do we need to to point that directly? shouldn't be the same version as from unicorn?

File: src/test/scala/org/virtuslab/beholder/form/FormFiltersTestForLargeElements.scala [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+        all <- allFromDb
+        orderByFirstField <- doFilters(bfd, baseFilter.copy(orderBy = Some(Order("f1", asc = false))))
+      } yield {
+        val fromDbOrderedByFirstField = all.sortBy(view => (view.f1, view.f1)).reverse
+        orderByFirstField should contain theSameElementsInOrderAs fromDbOrderedByFirstField
+      }
+    }
+  }
+
+  it should "take correctly" in { implicit f =>
```
Comment (JakubSroka) [inline src/test/scala/org/virtuslab/beholder/form/FormFiltersTestForLargeElements.scala:120]: maybe be more descriptive?

File: src/test/scala/org/virtuslab/beholder/form/FormFiltersTestForLargeElements.scala [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+  it should "filter by int field" in { implicit f =>
+    rollbackActionWithModel {
+      val bfd = new BaseFilterData
+      import bfd._
+      for {
+        all <- allFromDb
+        orderByIntField <- doFilters(bfd, baseFilter.copy(data = baseFilter.data.updated(2, Some(10))))
+      } yield {
+        val fromDbOrderedByIntField = all.filter(_.f2 == 2)
+        orderByIntField should contain theSameElementsInOrderAs fromDbOrderedByIntField.drop(1)
```
Comment (JakubSroka) [inline src/test/scala/org/virtuslab/beholder/form/FormFiltersTestForLargeElements.scala:157]: why is there drop(1) ?

**Review by JakubSroka — CHANGES_REQUESTED** [review]

File: src/test/scala/org/virtuslab/beholder/form/FormFiltersTestForLargeElements.scala (hunk 4 above)
Comment (agluszyk) [reply src/test/scala/org/virtuslab/beholder/form/FormFiltersTestForLargeElements.scala:157]: It's based on `FormFiltersTest`

File: src/test/scala/org/virtuslab/beholder/form/FormFiltersTestForLargeElements.scala (hunk 3 above)
Comment (agluszyk) [reply src/test/scala/org/virtuslab/beholder/form/FormFiltersTestForLargeElements.scala:120]: It's based on FormFiltersTest

File: build.sbt (hunk 1 above)
Comment (agluszyk) [reply build.sbt]: removed

Comment (agluszyk) [reply build.sbt]: removed

File: src/test/scala/org/virtuslab/beholder/form/FormFiltersTestForLargeElements.scala (hunk 2 above)
Comment (agluszyk) [reply src/test/scala/org/virtuslab/beholder/form/FormFiltersTestForLargeElements.scala:52]: done

**Review by fazzou — APPROVED** [review]

**Review by JakubSroka — APPROVED** [review]

**Pull request merged**